src/Microsoft.Azure.WebJobs.Logging/Internal/LogReader.cs

```diff
@@ -155,20 +184,186 @@ public LogReader(CloudTable table)
         {
             TableQuery<RecentPerFuncEntity> rangeQuery = RecentPerFuncEntity.GetRecentFunctionsQuery(queryParams);
 
-            CancellationToken cancellationToken;
-            TableContinuationToken realContinuationToken = Utility.DeserializeToken(continuationToken);
-            var segment = await _instanceTable.SafeExecuteQuerySegmentedAsync<RecentPerFuncEntity>(
-                rangeQuery, 
-                realContinuationToken, 
-                cancellationToken);
+            var iter = new EpochTableIterator(_tableLookup);
+            var results = await iter.SafeExecuteQuerySegmentedAsync<RecentPerFuncEntity>(rangeQuery, queryParams.Start, queryParams.End);
+
+            return results.As<IRecentFunctionEntry>();
+        }
+
+        // Helper to run queries which can span multiple tables.         
+        private class EpochTableIterator
+        {
+            private readonly ILogTableProvider _tableLookup;
+            private Dictionary<long, CloudTable> _tables; // map of epoch to physical tables.
+
+            public EpochTableIterator(ILogTableProvider tableLookup)
+            {
+                _tableLookup = tableLookup;
+            }
+            
+            private static async Task<Dictionary<long, CloudTable>> InitAsync(ILogTableProvider tableLookup)
+            {
+                Dictionary<long, CloudTable> d = new Dictionary<long, CloudTable>();
+
+                var tables = await tableLookup.ListTablesAsync();
+
+                foreach (var table in tables)
+                {
+                    var epoch = TimeBucket.GetEpochNumberFromTable(table);
+                    d[epoch] = table;
+                }
+                return d;
+            }
+
+            // Given an epoch, find the next one in physicalEpochs that it would fall into, using reverse-chronological order.              
+            // physicalEpochs - epochs for existing tables, sorted in ascending order.  This is generally short (expected under 50 elements)
+            // Returns a physical epoch (from the list passed in) that the requested epoch falls into 
+            // or -1 if there's no physical epoch below this.
+            private static long FindNext(long epoch, long[] physicalEpochs)
```

Comment:
what if we reversed the sorting order, can simplify this whole function to something like
`physicalEpochs.First((physical) => physical <= epoch)`